modules/volunteers/report/pages/body.inc

```diff
@@ -0,0 +1,181 @@
+          <button id='gen_der_csv' class='UI-eventbutton' onclick='generateDerivedCSV();'>Export Derived CSV</button>
+          <button id='gen_csv' class='UI-eventbutton' onclick='generateCSV();'>Export Raw CSV Table</button>
+        </div>
+      </div>
+    </div>
+
+    <div class="UI-container UI-center event-color-primary">
+      <div class='UI-tabletitle'>Hours</div>
+      <table class='UI-stripedtable' summary='Department Volunteer Summery'>
+        <tr>
+            <th id="Department">Department Worked</th>
+            <th id="Entries">Total Entries</th>
+            <th id="Volunteers">Total Volunteers</th>
+            <th id="Hours">Total Hours</th>
+        </tr>
+<?php hour_table();?>
+      </table>
+      <div class='UI-tablefooter'>&nbsp;</div>
+    </div>
+  </div>
+
+  <div class='UI-sidebar-hidden UI-fixed' id='department_report_div'>
+    <div class='UI-center'>
+      <h2 class='UI-red'>Department Report</h2>
+    </div>
+    <hr>
+    <div>
+      <h2 id='dept_name' class='event-color-secondary'></h2>
+      <input class="UI-hiddeninput" id="dept_data" readonly>
+      <input class="UI-hiddeninput" id="dept_data_name" readonly>
+    </div>
+    <hr>
+    <div class='UI-center'>
+      <button id='generate_dept_report' class='UI-eventbutton'
+          onclick='generateDeptReport();'>
+         Generate Report &lt;CSV&gt;
+      </button>
+      <button class='UI-redbutton'
+          onclick='hideSidebar();'>
+        Close
+      </button>
+    </div>
+  </div>
+
+  <div class='UI-sidebar-hidden UI-fixed' id='csv_export_div'>
+    <div class='UI-center'>
+      <h2 class='UI-red'>Export Raw CSV Data</h2>
+    </div>
+    <div class='UI-center'>
+      <label class='UI-label' for='csv_table'>
+      Table:</label>
+      <select class="UI-margin UI-select" style="width:auto" id="csv_table">
+        <option value=HourRedemptions>HourRedemptions</option>
+        <option value=VolunteerHours selected>VolunteerHours</option>
+        <option value=VolunteerRewards>VolunteerRewards</option>
+        <option value=RewardGroup>RewardGroup</option>
+      </select>
+    </div>
+    <div class='UI-center'>
+      <button id='generate_csv' class='UI-eventbutton'
+          onclick='generatCSVReport();'>
+         Generate CSV
+      </button>
+      <button class='UI-redbutton'
+          onclick='hideSidebar();'>
+        Close
+      </button>
+    </div>
+  </div>
+
+  <div class='UI-sidebar-hidden UI-fixed' id='csv_export_derived_div'>
+    <div class='UI-center'>
+      <h2 class='UI-red'>Export Derived CSV Data</h2>
+    </div>
+    <div class='UI-center'>
+      <label class='UI-label' for='der_csv_table'>
+      Table:</label>
+      <select class="UI-margin UI-select" style="width:auto" id="der_csv_table">
+        <option value="SELECT r.*, (SELECT COUNT(h.PrizeID) FROM HourRedemptions AS h WHERE h.PrizeID = r.PrizeID GROUP BY h.PrizeID) as Claimed FROM VolunteerRewards as r;">Volunteer Rewards</option>
```

Comment:
Um...let's NOT include SQL as possible values here. We do not want to get into the habit of accepting and executing SQL submitted by a form, ever.

Reply:
Ah yes... that is pre-existing for the record. Still horrible and needs to be fixed but pre-existing. :)

Reply:
addressed